### DISCUSSION TITLE
First ideas

### DISCUSSION THREAD

**Discussion by Hboni** [Hboni/francis#2] [Ideas]
The idea of this Francis project is to have a user interface that helps to create an image analysis pipeline, with the possibility to change parameters of processes on-the-go, and with a preview of images in the pipeline.

The first idea is to have an area to create pipelines using boxes to represent processes (import, erosion, ....), with the possibility to link different boxes to create pipelines. Each box has its own parameters which can be modified. A little window with the display of the image for each step next to each box help to follow pipeline creation.
It could also be possible to save the analysis pipeline to be reopened in Francis, to apply on different images for example.

Any other suggestions could be great!

**Comment by Hboni** [comment]
@jeaunrg The first version that you developed is awesome ! 

Do you think that's a good idea to be able to put process parameters (for example for erosion process) in the main window, below node name? I am not really comfort with all the new windows each time you click on process/visualization.
To be convenient, it can be also possible to hide those process parameters to not overcharged workspace interface.

What do you think? Do you have other propositions?

**Reply by jeaunrg** [reply]
Sorry I didnt see your message!
Its a good idea, it will visually simplify the interface, Im working on it.
